es/Grid/Grid.js

```diff
@@ -282,13 +282,19 @@ var Grid = (function (_Component) {
       var scrollToRow = _props2.scrollToRow;
       var width = _props2.width;
       var _state = this.state;
+      var isScrolling = _state.isScrolling;
       var scrollLeft = _state.scrollLeft;
       var scrollTop = _state.scrollTop;
 
       // Make sure any changes to :scrollLeft or :scrollTop get applied
-      if (scrollLeft >= 0 && scrollLeft !== prevState.scrollLeft || scrollTop >= 0 && scrollTop !== prevState.scrollTop) {
-        this.refs.scrollingContainer.scrollLeft = scrollLeft;
-        this.refs.scrollingContainer.scrollTop = scrollTop;
+      // Don't re-apply while a scroll is in progress; this causes slow scrolling for certain OS/browser combinations (eg. Windows and Firefox)
+      if (!isScrolling) {
```

Comment:
This sounds much better to me. But actually you just need to differentiate between two sources of scroll position changes: observation of browser scrolling, and intentional adjustments. So instead of isScrolling, I'd call that part of the state scrollPosUpdateReason, with the two possible values "observed" and "requested".

And in the comment, instead of "this causes slow scrolling for certain OS/browser combinations (eg. Windows and Firefox)" I'd write:
"Assigning to scrollLeft / scrollTop tells the browser to interrupt any running scroll animations, and to discard any pending async changes to the scroll position that may have happened in the meantime (e.g. on a separate scrolling thread). So we only set the element's scrollLeft/scrollTop when we really require an adjustment of the scroll position."

I found out that the old code altered scrolling behavior even on OS X with touchpad scrolling (i.e. without any scroll animations), just less noticeably so. What happened was that the scroll event fired, so the current scroll position was recorded in the state. But Grid's _setNextState processes state changes asynchronously, so before componentDidUpdate was called with that state, the scroll position could have changed again, and then we'd overwrite the new scroll position with the one that we recorded in the state.